views/middleware.go

```diff
@@ -14,46 +14,232 @@ import (
 
 // RequiresLogin is a middleware which will be used for each
 // httpHandler to check if there is any active session
-func (v *Views) RequiresLogin(h http.Handler) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		session, err := v.cookie.Get(r, v.conf.SessionCookieName)
+func (v *Views) RequiresLogin(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		session, err := v.cookie.Get(c.Request(), v.conf.SessionCookieName)
 		if err != nil {
 			log.Println(err)
-			http.Redirect(w, r, "/login", http.StatusFound)
-			return
+			return v.LoginFunc(c)
 		}
-		if !helpers.GetUser(session).Authenticated {
+		user1 := helpers.GetUser(session)
+		user2, err := v.user.GetUser(c.Request().Context(), user1)
```

Comment:
What's the difference between `user1` and `user2` (and could the variables be renamed so I don't have to ask that question)?

Reply:
I will find all things like that and rename them

Reply:
Changed the name